src/ORM/DataQuery.php

```diff
@@ -298,13 +298,19 @@ public function getFinalisedQuery($queriedColumns = null)
                             $collisionClassColumn = $schema->sqlColumnForField($collisionClass, 'ClassName');
                             $collisionClasses = ClassInfo::subclassesFor($collisionClass);
                             $collisionClassesSQL = implode(', ', Convert::raw2sql($collisionClasses, true));
-                            $caseClauses[] = "WHEN {$collisionClassColumn} IN ({$collisionClassesSQL}) THEN $collision";
+                            // Only add clause if this is already joined "Unknown column 'ClassName' error"
+                            $collisionTable = $schema->tableForField($collisionClass, 'ClassName');
```

Comment:
We probably shouldn't override the `$collisionTable` variable set for the whole loop, even though it's not used beyond this line, just as a good practice. Could be named e.g. `$tableForClassNameField` or `classNameFieldTable` or similar.